Building abseil on Linux m68k fails with error: static assertion failed: Platform is not 64-bit

Building abseil on Linux m68k fails with:

```
make[4]: Entering directory '/<<PKGBUILDDIR>>/static'
[ 23%] Building CXX object absl/base/CMakeFiles/malloc_internal.dir/internal/low_level_alloc.cc.o
cd "/<<PKGBUILDDIR>>/static/absl/base" && /usr/bin/c++  -I"/<<PKGBUILDDIR>>" -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -Wall -Wextra -Wcast-qual -Wconversion-null -Wmissing-declarations -Woverlength-strings -Wpointer-arith -Wundef -Wunused-local-typedefs -Wunused-result -Wvarargs -Wvla -Wwrite-strings -Wno-missing-field-initializers -Wno-sign-compare -DNOMINMAX -std=gnu++14 -o CMakeFiles/malloc_internal.dir/internal/low_level_alloc.cc.o -c "/<<PKGBUILDDIR>>/absl/base/internal/low_level_alloc.cc"
In file included from /<<PKGBUILDDIR>>/absl/base/internal/low_level_alloc.cc:26:
/<<PKGBUILDDIR>>/absl/base/internal/direct_mmap.h: In function 'void* absl::debian1::base_internal::DirectMmap(void*, size_t, int, int, int, off64_t)':
/<<PKGBUILDDIR>>/absl/base/internal/direct_mmap.h:127:39: error: static assertion failed: Platform is not 64-bit
  127 |   static_assert(sizeof(unsigned long) == 8, "Platform is not 64-bit");
      |                 ~~~~~~~~~~~~~~~~~~~~~~^~~~
make[4]: *** [absl/base/CMakeFiles/malloc_internal.dir/build.make:85: absl/base/CMakeFiles/malloc_internal.dir/internal/low_level_alloc.cc.o] Error 1
make[4]: Leaving directory '/<<PKGBUILDDIR>>/static'
make[3]: *** [CMakeFiles/Makefile2:816: absl/base/CMakeFiles/malloc_internal.dir/all] Error 2
make[3]: *** Waiting for unfinished jobs....
[ 23%] Building CXX object absl/time/CMakeFiles/time_zone.dir/internal/cctz/src/time_zone_posix.cc.o
cd "/<<PKGBUILDDIR>>/static/absl/time" && /usr/bin/c++  -I"/<<PKGBUILDDIR>>" -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -Wall -Wextra -Wcast-qual -Wconversion-null -Wmissing-declarations -Woverlength-strings -Wpointer-arith -Wundef -Wunused-local-typedefs -Wunused-result -Wvarargs -Wvla -Wwrite-strings -Wno-missing-field-initializers -Wno-sign-compare -DNOMINMAX -std=gnu++14 -o CMakeFiles/time_zone.dir/internal/cctz/src/time_zone_posix.cc.o -c "/<<PKGBUILDDIR>>/absl/time/internal/cctz/src/time_zone_posix.cc"
[ 24%] Building CXX object absl/time/CMakeFiles/time_zone.dir/internal/cctz/src/zone_info_source.cc.o
cd "/<<PKGBUILDDIR>>/static/absl/time" && /usr/bin/c++  -I"/<<PKGBUILDDIR>>" -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -Wall -Wextra -Wcast-qual -Wconversion-null -Wmissing-declarations -Woverlength-strings -Wpointer-arith -Wundef -Wunused-local-typedefs -Wunused-result -Wvarargs -Wvla -Wwrite-strings -Wno-missing-field-initializers -Wno-sign-compare -DNOMINMAX -std=gnu++14 -o CMakeFiles/time_zone.dir/internal/cctz/src/zone_info_source.cc.o -c "/<<PKGBUILDDIR>>/absl/time/internal/cctz/src/zone_info_source.cc"
[ 25%] Linking CXX static library libabsl_time_zone.a
cd "/<<PKGBUILDDIR>>/static/absl/time" && /usr/bin/cmake -P CMakeFiles/time_zone.dir/cmake_clean_target.cmake
cd "/<<PKGBUILDDIR>>/static/absl/time" && /usr/bin/cmake -E cmake_link_script CMakeFiles/time_zone.dir/link.txt --verbose=1
/usr/bin/ar qc libabsl_time_zone.a CMakeFiles/time_zone.dir/internal/cctz/src/time_zone_fixed.cc.o CMakeFiles/time_zone.dir/internal/cctz/src/time_zone_format.cc.o CMakeFiles/time_zone.dir/internal/cctz/src/time_zone_if.cc.o CMakeFiles/time_zone.dir/internal/cctz/src/time_zone_impl.cc.o CMakeFiles/time_zone.dir/internal/cctz/src/time_zone_info.cc.o CMakeFiles/time_zone.dir/internal/cctz/src/time_zone_libc.cc.o CMakeFiles/time_zone.dir/internal/cctz/src/time_zone_lookup.cc.o CMakeFiles/time_zone.dir/internal/cctz/src/time_zone_posix.cc.o CMakeFiles/time_zone.dir/internal/cctz/src/zone_info_source.cc.o
/usr/bin/ranlib libabsl_time_zone.a
make[4]: Leaving directory '/<<PKGBUILDDIR>>/static'
```

Full log available here: https://buildd.debian.org/status/fetch.php?pkg=abseil&arch=m68k&ver=0%7E20200923.3-1&stamp=1612822687&raw=0

Access to a Linux m68k environment can be obtained through QEMU: https://wiki.debian.org/M68k/QemuSystemM68k

--- a/absl/base/internal/direct_mmap.h
+++ b/absl/base/internal/direct_mmap.h
@@ -74,6 +74,7 @@ namespace base_internal {
 inline void* DirectMmap(void* start, size_t length, int prot, int flags, int fd,
                         off64_t offset) noexcept {
 #if defined(__i386__) || defined(__ARM_ARCH_3__) || defined(__ARM_EABI__) || \
+    defined(__m68k__) ||                                                     \
     (defined(__mips__) && _MIPS_SIM == _MIPS_SIM_ABI32) ||                   \
     (defined(__PPC__) && !defined(__PPC64__)) ||                             \
     (defined(__riscv) && __riscv_xlen == 32) ||                              \

--- a/absl/debugging/internal/examine_stack.cc
+++ b/absl/debugging/internal/examine_stack.cc
@@ -51,6 +51,8 @@ void* GetProgramCounter(void* vuc) {
 #elif defined(__i386__)
     if (14 < ABSL_ARRAYSIZE(context->uc_mcontext.gregs))
       return reinterpret_cast<void*>(context->uc_mcontext.gregs[14]);
+#elif defined(__m68k__)
+    return reinterpret_cast<void*>(context->uc_mcontext.gregs[16]);
 #elif defined(__mips__)
     return reinterpret_cast<void*>(context->uc_mcontext.pc);
 #elif defined(__powerpc64__)

